add doc & test for `useMountedRef` (#679)
Shopify/quilt: packages/react-hooks/README.md

  return null;
}
```

### `useMountedRef()`

> This hook keep track of a component's mounted / un-mounted status and returns a ref object like React’s [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) with a boolean value representing said status. This is often use when a component contains async task that set state after the task resolved.

Reviewer: ```suggestion
This hook returns a ref object (like React’s [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref)) that tracks whether a component is mounted. This is often use when a component contains async tasks that set state after the task resolved.
```

```tsx
import * as React from 'react';
import {useMountedRef} from '@shopify/react-hooks';

> const promise = new Promise();

Reviewer: Newline after imports

function MockComponent() {
  const [resolved, setResolved] = React.useState(false);
  const mounted = useMountedRef();

  async function handleOnClick() {

>     await promise;

Reviewer: I'd usually show a slightly more realistic example, like `await waitForSomethingToLoad()`
